File: 1033.判断闰年/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int year = scanner.nextInt();
        System.out.println(isLeapYear(year) ? "yes" : "no");
    }

    private static boolean isLeapYear(int year) {
        if (year % 4 == 0 && year % 100 != 0) {
            return true;
        } else if (year % 400 == 0) {
            return true;
        } else
            return false;
    }
}

File: 1033.判断闰年/main.cpp
#include <iostream>

using namespace std;

int main()
{
    int year;
    cin>>year;
    if(year%4 == 0)
    {
        if( year%100 == 0)
        {
            if ( year%400 == 0)
                cout<< "yes" <<endl;
            else
                cout<< "no" <<endl;
        }
        else
            cout<< "yes" <<endl;
    }
    else
        cout<< "no" <<endl;
    return 0;
}
